x-pack/plugins/security_solution/public/detection_engine/rule_creation/components/required_fields/make_validate_required_field.ts

```diff
@@ -51,3 +51,20 @@ export function makeValidateRequiredField(parentFieldPath: string) {
     }
   };
 }
+
+function getAllRequiredFieldsValues(
+  form: { getFields: FormHook['getFields'] },
+  parentFieldPath: string
+) {
+  /*
+    Getting values for required fields via flattened fields instead of using `getFormData`.
+    This is because `getFormData` applies a serializer function to field values, which might update values.
```

Comment:
[Serializer](https://docs.elastic.dev/form-lib/examples/serializers-deserializers) is a function you can pass to the form library. It converts form data from an internal format to an output format when you click Submit. 

To get the form data for validation, we normally use `form.getFormData()`. However, this method internally calls the serializer. In this case, we need the form data in its internal format, before it is serialized, because the returned value would be different if you pass a different serialzers.